### PR TITLE
Added Spinner widget

### PR DESCRIPTION
**Pull request by FloVanGH**

### CODE REVIEW

File: internal/compiler/widgets/common/blub.slint
Comment (tronical) [inline internal/compiler/widgets/common/blub.slint:1]: This looks like a file that was accidentally added?

File: internal/compiler/widgets/cupertino-base/spinner.slint [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+export component Spinner {
+    in property <float> progress <=> base.progress;
+    in property <bool> indeterminate <=> base.indeterminate;
+
+    min-width: 30px;
+    min-height: 30px;
+    horizontal-stretch: 0;
+    vertical-stretch: 0;
+    accessible-role: none;
+    accessible-value: root.progress;
```
Comment (tronical) [inline internal/compiler/widgets/cupertino-base/spinner.slint:16]: What is the meaning of `none` as accessible role but `value` being set to the progress?

Comment (tronical) [reply internal/compiler/widgets/cupertino-base/spinner.slint:16]: I see that this is copy & paste from the ProgressIndicator. Nevermind :)

Comment (tronical) [reply internal/compiler/widgets/cupertino-base/spinner.slint:16]: Fixed in https://github.com/slint-ui/slint/commit/a5c3692fbc14eca877b1ed2ad46fc6925eaddc69 ;-)

File: examples/gallery/ui/pages/controls_page.slint [hunk 2]
```diff
@@ -106,16 +106,29 @@ export component ControlsPage inherits Page {
...
         vertical-stretch: 0;
 
         VerticalLayout {
             spacing: 16px;
 
-            i-progress-indicator := ProgressIndicator {
-                min-width: 160px;
-                progress: (i-slider.value - i-slider.minimum) / (i-slider.maximum - i-slider.minimum);
-                indeterminate: true;
+            HorizontalBox {
```
Comment (tronical) [inline examples/gallery/ui/pages/controls_page.slint:115]: Olivier and I discussed and want to propose the following alternative.

Before:
Screenshot 2023-11-07 at 13 45 01


After:


Screenshot 2023-11-07 at 13 45 17

This is what's in the GroupBox:

```
        GridBox {
            spacing: 16px;

            i-progress-indicator := ProgressIndicator {
                row: 0;
                col: 0;
                min-width: 160px;
                progress: (i-slider.value - i-slider.minimum) / (i-slider.maximum - i-slider.minimum);
                indeterminate: true;
            }

            Rectangle {
                row: 0;
                col: 1;
                rowspan: 2;
                Spinner {
                    progress: i-progress-indicator.progress;
                    indeterminate: i-progress-indicator.indeterminate;
                }
            }

            CheckBox {
                row: 1;
                col: 0;
                text: @tr("indeterminate");
                checked <=> i-progress-indicator.indeterminate;
            }
        }
```

File: internal/compiler/widgets/common/blub.slint
Comment (FloVanGH) [reply internal/compiler/widgets/common/blub.slint:1]: yes that's true thanks

File: internal/compiler/widgets/cupertino-base/spinner.slint (hunk 1 above)
Comment (FloVanGH) [reply internal/compiler/widgets/cupertino-base/spinner.slint]: thx, is now used also for the spinner

File: examples/gallery/ui/pages/controls_page.slint (hunk 2 above)
Comment (FloVanGH) [reply examples/gallery/ui/pages/controls_page.slint]: agreed and changed, thx.

**Review by tronical — APPROVED** [review]

**Pull request merged**